python/graphstorm/model/lm_model/hf_bert.py

```diff
@@ -16,43 +16,32 @@
     Huggingface bert support
 """
 import time
-import inspect
 
 import numpy as np
 import torch as th
 from torch import nn
 
-from transformers import BertModel, BertConfig
+from transformers import AutoModel, AutoConfig, PreTrainedModel
 
 from .lm_model import TOKEN_IDX, ATT_MASK_IDX, VALID_LEN, TOKEN_TID_IDX
 from .lm_model import GSFLanguageModelWrapper
 
-def load_hfbert_model(bert_configs):
-    """ Load huggingface bert model
+def load_hfbert_model(bert_configs: dict):
+    """ Load huggingface LM model
     """
     model_name = bert_configs["model_name"]
-    gradient_checkpointing = bert_configs["gradient_checkpoint"]
-    signature = inspect.signature(BertConfig.__init__)
-    hidden_dropout_prob = bert_configs["hidden_dropout_prob"] \
-        if "hidden_dropout_prob" in bert_configs \
-        else signature.parameters['hidden_dropout_prob'].default
-    attention_probs_dropout_prob = bert_configs["attention_probs_dropout_prob"] \
-        if "attention_probs_dropout_prob" in bert_configs \
-        else signature.parameters['attention_probs_dropout_prob'].default
-    config = BertConfig.from_pretrained(model_name,
-                                        gradient_checkpointing=gradient_checkpointing,
-                                        hidden_dropout_prob=hidden_dropout_prob,
-                                        attention_probs_dropout_prob=attention_probs_dropout_prob)
-
-    lm_model = BertModel.from_pretrained(model_name, config=config)
+    config = AutoConfig.from_pretrained(
+        model_name, **{key: val for key, val in bert_configs.items() if key != 'model_name'})
```

Comment:
does this implementation lose any information?

Reply:
@classicsong can you help review?

Reply:
It won't.